ambry-store/src/main/java/com.github.ambry.store/StoreMetrics.java

```diff
@@ -143,7 +143,13 @@ public Long getValue() {
         return index.hardDeleteCaughtUp() ? 1L : 0L;
       }
     };
-    registry.register(MetricRegistry.name(PersistentIndex.class, name + "-hardDeleteCaughtUp"),
-        hardDeleteCaughtUp);
+    registry.register(MetricRegistry.name(PersistentIndex.class, name + ".HardDeleteCaughtUp"), hardDeleteCaughtUp);
+  }
+
+  static String getPartitionName(String replicaDir) {
```

Comment:
The flow is `StoreManager -> BlobStore -> StoreMetrics`, and `BlobStore` has no knowledge about `Replica` or `Partition`. So partition name should be passed all the way from `StoreManager` to `StoreMetrics`. @pnarayanan I guess this what you meant.